### PR TITLE
Tr 13 add new hosted field for card holder name

### PR DESCRIPTION
**Pull request by javiercrac**

### CODE REVIEW

**Comment by apetrovici** [comment]
The Card Holder Name from HF is not sent in the requests.

**Review by apetrovici — CHANGES_REQUESTED** [review]
The Card Holder Name from HF is not sent in the requests.

File: assets/frontend/js/globalpayments-secure-payment-fields.js [hunk 1]
```diff
@@ -580,6 +584,13 @@
 					target: helper.getSubmitButtonTargetSelector( this.id )
 				}
 			};
+			if ( 'globalpayments_gpapi' == this.id ) {
```
Comment (apetrovici) [inline assets/frontend/js/globalpayments-secure-payment-fields.js:587]: Please replace the condition with:
if (this.fieldOptions.hasOwnProperty('card-holder-name-field'))

This way, the code will work for other gateways adding the field.

File: src/Gateways/Requests/AbstractRequest.php [hunk 2]
```diff
@@ -61,7 +73,11 @@ public function get_default_args() {
 				'postalCode'     => $this->order->get_billing_postcode(),
 				'country'        => $this->order->get_billing_country(),
 			) : null,
-			RequestArg::CARD_HOLDER_NAME => null !== $this->order ? $this->order->get_formatted_billing_full_name() : null,
+			RequestArg::CARD_HOLDER_NAME =>
+				$this->get_card_holder_name(
+					null !== $this->order ? $this->order->get_formatted_billing_full_name() : null,
+					$this->data
```
Comment (apetrovici) [inline src/Gateways/Requests/AbstractRequest.php:79]: I still see the billing name in the requests, not the input from the hf.

**Review by apetrovici — CHANGES_REQUESTED** [review]

File: src/Gateways/Requests/AbstractRequest.php (hunk 2 above)
Comment (javiercrac) [reply src/Gateways/Requests/AbstractRequest.php]: The issue here is that there are many request running on here. To debug it, i had to log every request and then  i could change it to the right one and get the data .

File: assets/frontend/js/globalpayments-secure-payment-fields.js [hunk 3]
```diff
@@ -580,6 +584,13 @@
 					target: helper.getSubmitButtonTargetSelector( this.id )
 				}
 			};
+			if ( this.fieldOptions.hasOwnProperty( 'card-holder-name-field' ) ) {
+				fields["card-holder-name"] = {
+						placeholder: this.fieldOptions['card-holder-name-field'].placeholder,
```
Comment (ahumulescu) [inline assets/frontend/js/globalpayments-secure-payment-fields.js:589]: I think there might be some extra spaces in here. Can you check, please?

File: src/Gateways/Requests/AbstractRequest.php [hunk 4]
```diff
@@ -97,4 +114,5 @@ public function set_request_data ( array $data ) {
 			$this->data = $data;
 		}
 	}
+
```
Comment (ahumulescu) [inline src/Gateways/Requests/AbstractRequest.php:117]: I think this new line can be deleted.

**Review by ahumulescu — APPROVED** [review]

**Pull request merged**